src/lifetracker/calendar/CalendarListImpl.java

```diff
@@ -392,33 +404,49 @@ void filterByEndTime(TreeMap<Integer, CalendarEntry> treeMap, LocalTime endTime)
         }
         for (int i = treeMap.firstKey(); i < treeMap.lastKey() + 1; i++) {
```

Comment:
You can iterate through the map directly with a for loop
```Java
for(CalendarEntry c : treeMap){
    int  = c.getId();
}
```

or via the entry set 

```Java
for(Map.Entry<Integer, CalendarEntry> e: treeMap.entrySet){
   e.getKey();
   e.getValue();
}
```